lib/matplotlib/tests/test_widgets.py

```diff
@@ -1006,8 +1006,12 @@ def test_check_radio_buttons_image():
     rb = widgets.RadioButtons(rax1, ('Radio 1', 'Radio 2', 'Radio 3'))
     with pytest.warns(DeprecationWarning):
         rb.circles  # Trigger the old-style elliptic radiobuttons.
-    widgets.CheckButtons(rax2, ('Check 1', 'Check 2', 'Check 3'),
-                         (False, True, True))
+    cb = widgets.CheckButtons(rax2, ('Check 1', 'Check 2', 'Check 3'),
+                              (False, True, True))
+    with pytest.warns(DeprecationWarning):
+        # Trigger old-style Rectangle check boxes
+        cb.rectangles
+        cb.lines
```

Comment:
This line is never reached. You need to put in in a separate `warns()` context.

Reply:
The `cb.lines` line?

Reply:
Yes.

Reply:
Ah okay, thanks. I realized a funny thing then. Accessing rectangles is all that's required since that's all that changes from the older reference image with the rectangles. The line crosses and the scatter crosses are the exact same. Anyway, I've updated it, thanks!

Reply:
I think you'd need to create a new `CheckButtons` object to check that the `cb.lines` is falling back correctly.

Reply:
@QuLogic sorry, I was unavailable over the last week. This is to just check the fallback functionality or to also compare it with the reference image? Because if it is the former, then adding another test to just check the fallback would be better I believe.

Reply:
Okay, I changed things up a little, let me know if this makes sense. Earlier, only the property that was accessed (say `rectangles`) was being set, but if someone is working with that property, they would also expect the other one (here `lines`) to be set in a similar way. So now, whenever the property is accessed for the first time, it calls the other to set it. This way, there's consistency in the method used for drawing the boxes. As a consequence, this doesn't need an extra instance for checking fallback for `lines` as that gets checked by `rectangles` itself. 

If we want a test for the fallbacks, I can add 2 extra tests that just check those things.

Reply:
We should still check both, I think. Also, does accessing the other object trigger a second warning unnecessarily?

Reply:
You can wrap it in a `_api.suppress_matplotlib_deprecation_warning` context, or put the code in a separate private function that's called from both.

Reply:
@QuLogic I've updated and added 2 tests for checking the fallbacks. Let me know if this makes sense now!